### ISSUE TITLE
Switch network enhancements - a simpler (IMHO) alternative to the showChangeNetwork functionality

### ISSUE THREAD

**Issue by kazazor** [rsksmart/rLogin#268]
At the moment you have the `rLogin.showChangeNetwork()` functionality.
This dialog basically shows the available network ids (which for me as a user, doesn't say too much compared to the network name..) and lets me choose the selected one.

Today in most modern dapps, the dapp itself has a dropdown to switch the chain it works with.
This is in order to provide a much better UX as it's there at the front of the page and not in a dedicated dialog that requires more clicks.

Here is Uniswap for example (but really any multichain dapp has something similar):
![Screen Shot 2022-04-30 at 20 19 47](https://user-images.githubusercontent.com/1577655/166115737-5a24696e-15a7-4a95-b901-2bb93ca1e227.png)

That means that the dapp has already information on the chain the user would like to connect to.
So while rLogin has the function `showChangeNetwork`, IMHO a much more needed function is: `switchNetwork` / `changeNetwork` (I prefer `switch` as it's the terminology metamask uses.

While I can do it with the provider directly, IMHO it would be easier for the developers using rLogin to have such an easy function supported.

That way if someone would like to use the `showChangeNetwork`, they can still do it. My bet is that most dapps will need the `switchNetwork` more as if they support multichain (which due to multichain.org coming to RSK more and more dapps will need it), they most likely already know which network to switch to thanks to their network picker dropdown.

### Special case
What happens when I'm choosing a network to connect to (using the dapp's dropdown) and then I don't have the chain setup on my metamask/wallet?
Uniswap and all the major dapps already solve that as well!
![Screen Shot 2022-04-30 at 20 24 13](https://user-images.githubusercontent.com/1577655/166115894-2c7277af-aa52-4c45-b179-94161342f736.png)

When trying to switch the network, rLogin could also invoke the register new network details if needed.
@ilanolkies and I talked about why should rLogin do that instead of just using the provider by the developers. IMO the answer is - to make it easier for developers using rLogin. Instead of us need to handle these cases, it can be done once by rLogin and make everyone else's lives easier.

### Bottom line

* Add the `switchNetwork` method to rLogin
* This method should get the network full details (so it could handle registering the network in case needed). For example:
```js
const options = {
  chainId: 42161,
  networkName: "Arbitrum",
  networkRpcUrl: "....",
  nativeCurrencySymbol: "ETH",
  blockExplorerUrl: "https://arbiscan.io/"
};
this.rLogin.switchNetwork(options);
```

![Screen Shot 2022-04-30 at 20 26 48](https://user-images.githubusercontent.com/1577655/166115982-747b8e5d-53a7-490e-99a6-2f9db7c6e249.png)

**Comment by ilanolkies** [comment]
IMHO rLogin is not responsible of doing this. If you want to have a button that operates MetaMask directly you can add it without needing rLogin in the middle.

The idea of rLogin is to simplify some common UXs for the developer, that is why we have the possibility to switch networks when the selected one does not match the supported ones for the dapp within the login flow.

<img width="458" alt="Screen Shot 2022-05-24 at 01 06 14" src="https://user-images.githubusercontent.com/36084092/169946776-8f3233c7-f8c6-4778-aea6-362ce5be3475.png">

You can also trigger that modal by `rLogin.showChangeNetwork()`. The idea is that the dapp does not need to implement the UI for that

<img width="461" alt="Screen Shot 2022-05-24 at 01 14 40" src="https://user-images.githubusercontent.com/36084092/169947591-b8af2ff5-d366-4423-ac61-504eae5431e1.png">

We only implement this for RSK now with trusted public nodes infra. We can explore how to add an optional configuration to allow the dapp developer to add their RPC connections for Metamask to this UI
